src/engraving/libmscore/articulation.cpp

```diff
@@ -685,34 +645,38 @@ void Articulation::doAutoplace()
         double md = minDistance().val() * sp;
 
         SysStaff* ss = m->system()->staff(si);
-        RectF r = bbox().translated(chordRest()->pos() + m->pos() + s->pos() + pos());
-
-        double d;
-        bool above = up();     // (anchor() == ArticulationAnchor::TOP_STAFF || anchor() == ArticulationAnchor::TOP_CHORD);
-        SkylineLine sk(!above);
-        if (above) {
-            sk.add(r.x(), r.bottom(), r.width());
-            d = sk.minDistance(ss->skyline().north());
-        } else {
-            sk.add(r.x(), r.top(), r.width());
-            d = ss->skyline().south().minDistance(sk);
-        }
 
-        if (d > -md) {
-            double yd = d + md;
+        Shape thisShape = shape();
+        for (ShapeElement& shapeEl : thisShape) {
+            RectF r = shapeEl.translated(chordRest()->pos() + m->pos() + s->pos() + pos());
```

Comment:
This should be called outside of the for loop

`chordRest()->pos() + m->pos() + s->pos() + pos()`